### PR TITLE
prevent creating cudnnPoolingOp with unsupported Pooling method

### PR DESCRIPTION
**Pull request by Godricly**
calling default sumpooling if cudnn is specified. #1984

### CODE REVIEW

File: src/operator/pooling.cu [hunk 1]
```diff
@@ -15,7 +15,17 @@ namespace op {
...
-  return new CuDNNPoolingOp(param);
+  switch (param.pool_type) {
+    case pool_enum::kMaxPooling:
+      return new CuDNNPoolingOp(param);
+    case pool_enum::kAvgPooling:
+      return new CuDNNPoolingOp(param);
+    case pool_enum::kSumPooling:
+      return new PoolingOp<gpu, mshadow::red::sum>(param);
+    default:
+      LOG(FATAL) << "unknown activation type";
```
Comment (piiswrong) [inline src/operator/pooling.cu]: pooling type

File: src/operator/pooling.cu [hunk 2]
```diff
@@ -15,7 +15,17 @@ namespace op {
...
 Operator *CreateOp<gpu>(PoolingParam param) {
 #if MXNET_USE_CUDNN == 1
-  return new CuDNNPoolingOp(param);
+  switch (param.pool_type) {
+    case pool_enum::kMaxPooling:
+      return new CuDNNPoolingOp(param);
+    case pool_enum::kAvgPooling:
+      return new CuDNNPoolingOp(param);
+    case pool_enum::kSumPooling:
+      return new PoolingOp<gpu, mshadow::red::sum>(param);
```
Comment (piiswrong) [inline src/operator/pooling.cu]: maybe give a warning for fallback

File: src/operator/pooling.cu (hunk 1 above)
Comment (Godricly) [inline src/operator/pooling.cu]: Whats wrong with pooling type?

File: src/operator/pooling.cu (hunk 2 above)
Comment (Godricly) [inline src/operator/pooling.cu]: a LOG(WARNING) is added. Should I also change cudnnpooling? There is a with-case checking pooling_enum too.

File: src/operator/pooling.cu (hunk 1 above)
Comment (piiswrong) [inline src/operator/pooling.cu]: The warning should read unknown pooling type

File: src/operator/pooling.cu [hunk 3]
```diff
@@ -15,7 +15,18 @@ namespace op {
...
+  switch (param.pool_type) {
+    case pool_enum::kMaxPooling:
+      return new CuDNNPoolingOp(param);
+    case pool_enum::kAvgPooling:
+      return new CuDNNPoolingOp(param);
+    case pool_enum::kSumPooling:
+      LOG(WARNING)<<"Sum Pooling is not supported by cudnn, MxNet Sum Pooling is applied.";
+      return new PoolingOp<gpu, mshadow::red::sum>(param);
+    default:
+      LOG(FATAL) << "unknown activation type";
```
Comment (piiswrong) [inline src/operator/pooling.cu]: LOG(FATAL) << "unknown activation type"; -> LOG(FATAL) << "unknown pooling type";

File: src/operator/pooling.cu (hunk 2 above)
Comment (Godricly) [inline src/operator/pooling.cu]: Ah..It is changed. I also checked pooling.cc file. The type appears there too. Can you fix that one? I don't know how to add that one into this.

**Comment by tqchen** [comment]
please fix the lint error and it can be merged in, thanks!

**Comment by Godricly** [comment]
I'll check this later tonight. The connection is not good.

**Pull request merged**